client/ayon_core/hosts/substancepainter/plugins/load/load_mesh.py

```diff
@@ -24,49 +163,36 @@ class SubstanceLoadProjectMesh(load.LoaderPlugin):
     order = -10
     icon = "code-fork"
     color = "orange"
+    project_templates = []
 
-    options = [
-        qargparse.Boolean(
-            "preserve_strokes",
-            default=True,
-            help="Preserve strokes positions on mesh.\n"
-                 "(only relevant when loading into existing project)"
-        ),
-        qargparse.Boolean(
-            "import_cameras",
-            default=True,
-            help="Import cameras from the mesh file."
-        )
-    ]
-
-    def load(self, context, name, namespace, data):
+    def load(self, context, name, namespace, options=None):
 
         # Get user inputs
-        import_cameras = data.get("import_cameras", True)
-        preserve_strokes = data.get("preserve_strokes", True)
+        result = SubstanceProjectConfigurationWindow.prompt(
+            self.project_templates).get_result()
```

Comment:
This _may_ fail. The `get_result` command gets the values from the Qt widgets after the dialog is closed - the closing of the dialog MAY delete the qt widgets on the C++ side (if not, then we have a memory leak?)

So technically, we must ensure we query the values from the QtWidgets before they get deleted - which as far as I know means getting it consistently before closing to the dialog. Or set the dialog to not delete on close automatically (which may be the default behavior, but then we currently have the memory leak where we are just generating more dialogs over time in memory all the time and never deleting the previous dialogs) and then get the values and then explicitly mark the dialog for deletion, e.g. `dialog.deleteLater()` or whatever.

Reply:
I have added` dialog.deleteLater()` after `dialog.exec_()` but not sure if it is the best way to implement it right after the dialog execution.